scripts/can_errgo/error_tui.go

```diff
@@ -0,0 +1,668 @@
+			if m.table.Focused() {
+				m.table.Blur()
+			} else {
+				m.table.Focus()
+			}
+		case "q", "ctrl+c":
+			return m, tea.Quit
+		case "a":	
+			
+			// Prevent panic if no row is selected
+			if m.table.SelectedRow() == nil{
+				m.table.SetCursorAndViewport(0)
+				return m, nil
+			}
+
+			// Remove the row and reset its count.
+			m.hiddenCounts[m.table.SelectedRow()[0]] = 0
+			newRows := deleteElementRow(m.table.Rows(), m.table.Cursor())
+			m.table.SetRows(newRows)
+
+			// May be unnecessary, but just in case we update the cursor.
+			if m.table.Cursor() == 0{
+				m.table.MoveDown(1)
+			}else{
+				m.table.MoveUp(1)
+			}
+			return m, nil
+		case "i":
+
+			// Prevent panic if no row is selected
+			if m.table.SelectedRow() == nil{
+				m.table.SetCursorAndViewport(0)
+				return m, nil
+			}
+
+			// Toggle the mask for the selected error and move it to the ignore menu.
+			m.ignoreMask = toggleBit(m.ignoreMask, m.errorToBit[m.table.SelectedRow()[0]] )
+			newSubRow := append(m.submenuTable.Rows(), table.Row{m.table.SelectedRow()[0]})
+			m.submenuTable.SetRows(newSubRow)
+			m.submenuTable.UpdateViewport()
+
+			// Remove the row from the main table and reset its count.
+			m.hiddenCounts[m.table.SelectedRow()[0]] = 0
+			newRows := deleteElementRow(m.table.Rows(), m.table.Cursor())
+			m.table.SetRows(newRows)
+
+			// May be unnecessary, but just in case we update the cursor.
+			m.table.MoveDown(1)
+
+			return m, nil
+        case "s":
+            if m.submenuActive {
+                // Hide submenu and return to main menu
+                m.submenuActive = false
+                m.table.Focus()
+            } else {
+                // Show submenu and focus it
+                m.submenuActive = true
+                m.table.Blur()
+                m.submenuTable.Focus()
+				m.table.SetShowCursor(false)
+            }
+		case "up":
+			m.table.MoveUp(1)
+			return m,nil	
+		case "down":
+			m.table.MoveDown(1)
+			return m,nil			
+	}
+	case CANMsg:
+		// Reset the timeout flag and update the last CAN message time
+		m.lastCANTime = time.Now()
+		m.isTimeout = false
+		// Perform bitwise AND to filter out the bits that are not of interest
+		msg.Value = msg.Value & m.ignoreMask
+		// Update the freshness of the errors
+		m.re_fresh()
+
+
+		for k := 0; k < 64; k++ { // Iterate through all 64 bits
+			if msg.Value&(1<<k) != 0 { // Check if the k-th bit is set
+
+				var errorNumberStr string = strconv.Itoa(k)
+
+				// Instead of "error" + #, this would be what the DBC file should map to, for now keep this for testing
+				if val, ok := m.hiddenCounts["error" + errorNumberStr]; ok{
+					m.hiddenCounts["error" + errorNumberStr ] = val + 1
+					
+				} else{
+					m.hiddenCounts["error" + errorNumberStr] = 1
+					m.errorToBit["error" + errorNumberStr] = k
+				}
+
+				// If the error is not in the table, add it, otherwise update the count.
+				index := findRowString(m.table.Rows(),  "error" + errorNumberStr)
```

Comment:
see file comment. Data should not be stored in the table

Reply:
Refactored